velox/functions/prestosql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -485,6 +485,128 @@ TEST_F(DateTimeFunctionsTest, yearTimestampWithTimezone) {
           "year(c0)", std::nullopt, std::nullopt));
 }
 
+TEST_F(DateTimeFunctionsTest, week) {
+  const auto week = [&](StringView date) {
+    return evaluateOnce<int64_t>(
+        "week(c0)",
+        std::make_optional(Timestamp(util::fromTimeString(date), 0)));
+  };
+
+  EXPECT_EQ(1, week("00:00:00"));
+  EXPECT_EQ(49, week("11:59:59"));
+  EXPECT_EQ(22, week("06:01:01"));
+  EXPECT_EQ(52, week("12:00:01"));
+
+  setQueryTimeZone("Pacific/Apia");
+  EXPECT_EQ(53, week("00:00:00"));
+  EXPECT_EQ(49, week("11:59:59"));
+  EXPECT_EQ(22, week("06:01:01"));
+  EXPECT_EQ(52, week("12:00:01"));
+}
+
+TEST_F(DateTimeFunctionsTest, weekDate) {
+  const auto week = [&](std::optional<Date> date) {
+    return evaluateOnce<int64_t>("week(c0)", date);
+  };
+  auto parseDate = [&](const std::string& dateStr) {
+    Date returnDate;
+    parseTo(dateStr, returnDate);
+    return returnDate;
+  };
+
+  EXPECT_EQ(53, week(parseDate("1919-12-31")));
```

Comment:
Please try some corner cases like a really small date like "0001 - 01- 01" and very large dates like "9999-12-31"